### DISCUSSION TITLE
Regain Space After Removing Files

### DISCUSSION THREAD

**Discussion by blueray453** [borgbackup/borg#8201] [Q&A] [answered]
Suppose i have removed a lot of files from /specific/folder/location. now i want to regain space. so i do the following. 
```
borg recreate . --exclude /specific/folder/location
borg compact .
borg create ....
```
Is this an acceptable approach.

Is there any easier way to do this?

**Comment by ThomasWaldmann** [comment, marked as answer]
> Suppose i have removed a lot of files from /specific/folder/location. 
> now i want to regain space. so i do the following.
> 
> borg recreate . --exclude /specific/folder/location
> borg compact .
> 
> Is this an acceptable approach?

Yes (I did not check the precise syntax and matching though).

To prevent unintended data loss, you should always add --dry-run --list 
to borg recreate and make very sure (by checking the output) that it 
really would do what you want it to do (and not e.g. exclude files that 
you wanted to keep).

After you made sure, remove the --dry-run and run recreate again, so it 
actually recreates the archives.

> Is there any easier way to do this?

Not if you want to keep all these archives.

If you do not need all of them, you could also first delete the unwanted 
ones and then recreate the wanted ones.


-- 

GPG Fingerprint: 6D5B EF9A DD20 7580 5747  B70F 9F88 FB52 FAF7 B393

**Comment by blueray453** [comment]
I mean after deleting large files, there is still files in /specific/folder/location

I just want to keep one copy in all archives where it will keep the current state of /specific/folder/location

In my case: `borg recreate . --exclude /specific/folder/location`  totally remove the folder then `borg create`  add the current state.

Is there any command that will remove (from all archives) only the files that are not present in /specific/folder/location from /specific/folder/location.

**Reply by ThomasWaldmann** [reply]
borg is deduplicating file chunks, so it always has only 1 copy of a specific chunk in the repository - you do not need to care for that manually.

an archive contains the state of the archived files at the point in time when they were archived.

and no, there is no such command. you need to say which files/folders you want removed (you want to exclude). maybe you can do that with some scripting, generating include/exclude rules for borg.
